#!/usr/bin/env python
PACKAGE = "ainstein_radar_filters"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("print_debug", bool_t, 0, "Enable debug message printing", False)
gen.add("print_timestamp", bool_t, 0, "Enable message timestamp printing", False)
gen.add("low_range_limit_m", double_t, 0, "Low range limit for point cloud debug messages (m)", 0.2, 0.0, 25)
gen.add("high_range_limit_m", double_t, 0, "High range limit for point cloud debug messages (m)", 3.0, 0.0, 25)
gen.add("low_speed_limit_mps", double_t, 0, "Low speed limit for point cloud debug messages (m/s)", -5.0, -10, 10)
gen.add("high_speed_limit_mps", double_t, 0, "High speed limit for point cloud debug messages (m/s)", 5.0, -10, 10)
gen.add("low_azimuth_limit_deg", double_t, 0, "Low azimuth limit for point cloud debug messages (deb)", -60, -90, 90)
gen.add("high_azimuth_limit_deg", double_t, 0, "High azimuth limit for point cloud debug messages (deb)", 60, -90, 90)
gen.add("low_elevation_limit_deg", double_t, 0, "Low elevation limit for point cloud debug messages (deb)", -60, -90, 90)
gen.add("high_elevation_limit_deg", double_t, 0, "High elevation limit for point cloud debug messages (deb)", 60, -90, 90)
gen.add("low_power_limit", double_t, 0, "Low power limit for point cloud debug messages (deb)", 0, 0, 65535)
gen.add("high_power_limit", double_t, 0, "High elevation limit for point cloud debug messages (deb)", 65535, 0, 65535)


exit(gen.generate(PACKAGE, "ainstein_radar_filters", "PointCloudDebug"))
